commit 1ceaa83a844cd3ecef25279d60720f910b96f297
Author: Cosmin Truta <ctruta@gmail.com>
Date:   Tue Sep 4 00:53:38 2018 -0400

    Fix the build with MSVC ARM64

Index: libpng1.6/arm/filter_neon_intrinsics.c
===================================================================
--- libpng1.6.orig/arm/filter_neon_intrinsics.c
+++ libpng1.6/arm/filter_neon_intrinsics.c
@@ -19,9 +19,9 @@
 #if PNG_ARM_NEON_IMPLEMENTATION == 1 /* intrinsics code from pngpriv.h */
 
 #if defined(_MSC_VER) && defined(_M_ARM64)
-#  include <arm64_neon.h>
+#include <arm64_neon.h>
 #else
-#  include <arm_neon.h>
+#include <arm_neon.h>
 #endif
 
 /* libpng row pointers are not necessarily aligned to any particular boundary,
Index: libpng1.6/arm/palette_neon_intrinsics.c
===================================================================
--- libpng1.6.orig/arm/palette_neon_intrinsics.c
+++ libpng1.6/arm/palette_neon_intrinsics.c
@@ -1,7 +1,5 @@
-
 /* palette_neon_intrinsics.c - NEON optimised palette expansion functions
  *
- * Copyright (c) 2018 Cosmin Truta
  * Copyright (c) 2017-2018 Arm Holdings. All rights reserved.
  * Written by Richard Townsend <Richard.Townsend@arm.com>, February 2017.
  *
@@ -14,11 +12,7 @@
 
 #if PNG_ARM_NEON_IMPLEMENTATION == 1
 
-#if defined(_MSC_VER) && defined(_M_ARM64)
-#  include <arm64_neon.h>
-#else
-#  include <arm_neon.h>
-#endif
+#include <arm_neon.h>
 
 /* Build an RGBA palette from the RGB and separate alpha palettes. */
 void
Index: libpng1.6/pngrtran.c
===================================================================
--- libpng1.6.orig/pngrtran.c
+++ libpng1.6/pngrtran.c
@@ -19,14 +19,10 @@
 #include "pngpriv.h"
 
 #ifdef PNG_ARM_NEON_IMPLEMENTATION
-#  if PNG_ARM_NEON_IMPLEMENTATION == 1
-#    define PNG_ARM_NEON_INTRINSICS_AVAILABLE
-#    if defined(_MSC_VER) && defined(_M_ARM64)
-#      include <arm64_neon.h>
-#    else
-#      include <arm_neon.h>
-#    endif
-#  endif
+#if PNG_ARM_NEON_IMPLEMENTATION == 1
+#define PNG_ARM_NEON_INTRINSICS_AVAILABLE
+#include <arm_neon.h>
+#endif
 #endif
 
 #ifdef PNG_READ_SUPPORTED
